nixos/modules/services/web-apps/peertube.nix

```diff
@@ -0,0 +1,357 @@
+    PrivateUsers = true;
+    ProtectClock = true;
+    ProtectHostname = true;
+    ProtectKernelLogs = true;
+    ProtectKernelModules = true;
+    ProtectKernelTunables = true;
+    ProtectControlGroups = true;
+    RestrictNamespaces = true;
+    LockPersonality = true;
+    RestrictRealtime = true;
+    RestrictSUIDSGID = true;
+    PrivateMounts = true;
+    # System Call Filtering
+    SystemCallArchitectures = "native";
+  };
+
+in {
+  options.services.peertube = {
+    enable = lib.mkEnableOption "Enable Peertube’s service";
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = "peertube";
+      description = "User account under which Peertube runs";
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      default = "peertube";
+      description = "Group under which Peertube runs";
+    };
+
+    hostname = lib.mkOption {
+      type = lib.types.str;
+      default = "example.com";
+      description = "Server name of reverse proxy";
+    };
+
+    listenHttp = lib.mkOption {
+      type = lib.types.int;
+      default = 9000;
+      description = "listen port for HTTP server";
+    };
+
+    listenWeb = lib.mkOption {
+      type = lib.types.int;
+      default = 443;
+      description = "listen port for WEB server";
+    };
+
+    enableWebHttps = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = "Enable or disable HTTPS protocol";
+    };
+
+    extraConfig = lib.mkOption {
+      type = lib.types.lines;
+      default = "";
+      example = ''
+        listen:
+          hostname: '0.0.0.0'
+        trust_proxy:
+          - '192.168.10.21'
+        log:
+          level: 'debug'
+      '';
+      description = "Extra config options for peertube";
+    };
+
+    redis = {
+      createLocally = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = "Configure local Redis server for PeerTube.";
+      };
+
+      host = lib.mkOption {
+        type = lib.types.str;
+        default = "127.0.0.1";
+        description = "Redis host.";
+      };
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        default = 6379;
+        description = "Redis port.";
+      };
+
+      passwordFile = lib.mkOption {
+        type = lib.types.nullOr lib.types.path;
+        default = null;
+        example = "/run/keys/peertube/password-redis-db";
+        description = "Password for redis database";
+      };
+
+      enableUnixSocket = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = "Use Unix socket";
+      };
+    };
+
+    database = {
+      createLocally = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = "Configure local PostgreSQL database server for PeerTube.";
+      };
+
+      host = lib.mkOption {
+        type = lib.types.str;
+        default = "/run/postgresql";
+        example = "192.168.15.47";
```

Comment:
```suggestion
        example = "/run/postgresql";
```

Reply:
Why write the same value in the parameter `example `and `default`?

Reply:
I thought they are not both shown on search.nixos.org but they are.